Extension/lib/filter/filters.js

```diff
@@ -433,7 +438,12 @@ RequestFilter.prototype = {
         if (this.requestCacheSize > this.requestCacheMaxSize) {
             this._clearRequestCache();
         }
-        this.requestCache[requestUrl] = [rule, refHost, requestType];
+
+        if (!this.requestCache[requestUrl]) {
+            this.requestCache[requestUrl] = Object.create(null);
+        }
+
+        this.requestCache[requestUrl][requestType] = [rule, refHost];
         this.requestCacheSize++;

```

Comment:
You don't need to increment cache size each time. You've already known that key is in the cache or not.